explore/src/main/scala/explore/itc/ItcColumn.scala

```diff
@@ -4,43 +4,26 @@
 package explore.itc
 
 import cats.Eq
+import cats.derived.*
 import cats.syntax.all._
 import eu.timepit.refined.auto._
 
 import scala.concurrent.duration._
 
-sealed trait ItcQueryProblems extends Product with Serializable
-
-object ItcQueryProblems {
-  case object UnsupportedMode          extends ItcQueryProblems
-  case object MissingWavelength        extends ItcQueryProblems
-  case object MissingSignalToNoise     extends ItcQueryProblems
-  case object MissingTargetInfo        extends ItcQueryProblems
-  case class GenericError(msg: String) extends ItcQueryProblems
-
-  implicit val eq: Eq[ItcQueryProblems] = Eq.instance {
-    case (UnsupportedMode, UnsupportedMode)           => true
-    case (MissingWavelength, MissingWavelength)       => true
-    case (MissingSignalToNoise, MissingSignalToNoise) => true
-    case (MissingTargetInfo, MissingTargetInfo)       => true
-    case (GenericError(a), GenericError(b))           => a === b
-    case _                                            => false
-  }
+enum ItcQueryProblems derives Eq {
+  case UnsupportedMode
+  case MissingWavelength
+  case MissingSignalToNoise
+  case MissingTargetInfo
+  case GenericError(msg: String)
 }
 
-sealed trait ItcResult extends Product with Serializable
+sealed trait ItcResult extends Product with Serializable derives Eq
```

Comment:
This could be converted to an `enum` as well. But it appears that enum instances can't have bodies. So `duration` in `Result` below would need to be an extension method. This makes it a lot less clear of a improvement.